packages/services/src/services/Translation/__tests__/Translation.test.js

```diff
@@ -76,7 +100,43 @@ describe('TranslationAPI', () => {
         origin: 'https://ibm.com',
       },
     });
-
     expect(response).toEqual(responseSuccess);
   });
+
+  it('should return a json with a recent timestamp', async () => {
+    // using very old cached session
+    sessionStorageMock.setItem(
+      'dds-translation-us-en',
+      JSON.stringify(oldSession)
+    );
+    const previousSession = JSON.parse(
+      sessionStorageMock.getItem('dds-translation-us-en')
+    );
+
+    // reinitializing import
+    const TranslationAPI = (await import('../Translation')).default;
+
+    const response = await TranslationAPI.getTranslation({
+      lc: 'en',
+      cc: 'us',
+    });
+
+    const newSession = JSON.parse(
+      sessionStorageMock.getItem('dds-translation-us-en')
+    );
+
+    // newest response and storage data should match
+    expect(response).toEqual(newSession);
+
+    // should contain timestamp
+    expect(response).toHaveProperty('timestamp');
+
+    // should not equal old timestamp
+    expect(previousSession.timestamp).not.toEqual(response.timestamp);
+
+    // timestamps should have at least a two hour difference
+    const timeDiff = response.timestamp - previousSession.timestamp,
+      _twoHours = 60 * 60 * 2000;
```

Comment:
Would you explain what makes the cache elapsed time here greater than two hours?

Reply:
Sure! As an example, the cached time in oldResponse is `1337` (a very old timestamp), while `response` would have the current time as `1603206344`. By doing `1603206344 - 1337` we receive `160319297`, which is greater than `7200000` (two hours). This is to ensure that the cache busting code runs for this test, as it is required for the time difference to be greater than two hours. 

If the previous session timestamp was much closer in time to the current one, the difference would be less than two hours, and the `response` received would remain unchanged from the previous session.

Reply:
Thank you for your answer @IgnacioBecerra! I was asking where in the code or in test that makes `timeDiff` greater than two hours, given the test says "`timeDiff` should be greater than two hours". What code sets `response.timestamp` and what code sets `previousSession.timestamp`? Did you make the timestamp deterministic by e.g. mocks (rather than relying on system time that is non-deterministic)?

Reply:
Ah I see what you mean! I set `previousSession` in line 9 using `import oldSession from './data/timestamp_response.json';`. 

`timestamp_response.json` uses the same data in `./data/response.json`, with the only difference is that it includes `timestamp: 1337`. This would be the only deterministic timestamp. On the other hand, `response` always contains the current timestamp from the latest response received from `TranslationAPI`.

Reply:
Cool thanks @IgnacioBecerra! Given your response, given we know the timestamps come from mock, why do we need to test the two timestamps between the two mocks? Or was it simply for ensuring we are picking up the right (fresh) data?

Reply:
It's the latter, I wanted to make extra sure the logic was working. Since we tested the fresh data in `sessionStorage` was the same as in the `response`, I used the mock to compare times, and had to check if `timeDiff` was greater than two hours.

Reply:
I see, thank you for clarifying @IgnacioBecerra! I suggest more direct way for testing if we pick up the right data. One approach can be object ref. Another approach may be unique key/ID.